### PR TITLE
feat(knowledge): zettel revision-keyed identity + Fleet share schema

### PR DESCRIPTION
**Pull request by H31MDALLR**
## Summary

Foundation for **miniforge-fleet's Phase E.3** (`share-learning` upload path). Closes the OSS-side prerequisites captured in the [fleet repo's Phase E planning doc](https://github.com/miniforge-ai/miniforge-fleet/blob/main/docs/pull-requests/2026-05-04-chris-phase-e-planning.md) as items #6 + #7 + #8 — schema additions, revision-keyed identity, and version provenance.

## Schema additions

All OPTIONAL — legacy file-backed stores round-trip without migration.

| Field | Type | Source | Why |
|---|---|---|---|
| `:zettel/revision-id` | UUID | Decision 6 | Immutable revision identity. Trust attaches here, not to the mutable `:zettel/id`. |
| `:zettel/digest` | 64-char SHA-256 hex | Decision 6 | Content fingerprint over the canonical-EDN of the content-bearing subset. |
| `:fleet/shareable` | boolean | Decision 8 | Producer's intent to share. Fleet's E.2 scanner has veto power when true. |
| `:fleet/share-scope` | `:org` / `:team` / `:repo` / `:workflow` | Decision 8 + 14 | Audience scope. Required from day one even on single-org deployments — Phase F multi-tenant retrofit otherwise becomes unbounded work. |
| `:privacy/classification` | `:public-org` / `:internal` / `:restricted` / `:secret` | Decision 8 | Authoritative privacy class. Embeddings-only sharing on `:restricted` / `:secret` content requires a Phase D approvals queue reference Fleet-side. |
| `:fleet/oss-version` | string | Decision 13 | Pin to the OSS version that captured the zettel. Fleet's E.4 quarantine + E.9 migration registry key off this. |

## `zettel.clj` changes

- **`compute-digest`** — pure SHA-256 over the canonical-EDN of the content-bearing subset (`:zettel/uid`, `:zettel/title`, `:zettel/content`, `:zettel/type`, `:zettel/dewey`, `:zettel/tags`, `:zettel/links`, `:zettel/source`). Operational metadata (`:fleet/*`, `:privacy/*`, `:zettel/created`, `:zettel/modified`, `:zettel/author`, `:zettel/backlinks`, `:fleet/oss-version`) is intentionally excluded — a privacy reclassification must NOT silently mint a new revision.

- **`revision-id-from-digest`** — `UUID/nameUUIDFromBytes` over the digest. Same content → same revision-id, the property miniforge-fleet's E.1 idempotent-retry path needs.

- **`create-zettel`** — gained kwargs for the new fields; now stamps `:zettel/digest` + `:zettel/revision-id` automatically.

- **`update-zettel`** — rotates digest + revision-id when `changes` touches a content-bearing field; leaves the revision intact when only operational metadata changes (so trust survives a classification edit).

## Tests

`zettel-revision-test.clj` — 14 tests, 41 assertions:
- revision-id is deterministic over content
- distinct content → distinct revision-id
- content edits rotate the revision
- operational-metadata edits preserve the revision
- new schema fields accept their documented enums + reject out-of-band values
- legacy zettels (without the new fields) still validate

## Test plan

- [x] `clj-kondo --lint components/knowledge` → 0 errors, 0 warnings
- [x] `clj -M:poly test brick:knowledge` → all green (52 tests, 223 assertions across the brick)
- [x] Pre-commit `bb pre-commit` (full sweep) → 5407 tests, 24200 passes, 0 failures
- [ ] CI green

## Out of scope (queued for follow-up OSS PRs)

- **Pack manifests as content-addressed `(zettel-id, revision-id, digest)` triples** (Decision 6 packs). Lands as a separate PR; this PR provides the revision-id triple it'll reference.
- **`:org/id` + `:workspace/id` + `:auth/context` propagation in agent-runtime events** (Decision 14). Separate PR.
- **Outbox event emission** (`zettel.promoted`, `pattern.applied`, `pattern.outcome`, `pattern.erased`). Separate PR; the Fleet side already consumes via E.1's event log.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: components/knowledge/src/ai/miniforge/knowledge/zettel.clj [hunk 1]
```diff
@@ -39,39 +105,73 @@
    - content - Markdown content body
    - type    - Zettel type keyword
 
-   Options:
+   Options (local Zettelkasten):
    - :dewey  - Dewey classification code (e.g., '210')
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/knowledge/src/ai/miniforge/knowledge/zettel.clj]: Already addressed by Copilot Autofix in commit 84706fba — the docstring example now reads `:dewey "210"` (string) instead of `:dewey '210'` (symbol). Resolving.

File: components/knowledge/src/ai/miniforge/knowledge/zettel.clj [hunk 2]
```diff
@@ -39,39 +105,73 @@
...
   [zettel changes]
-  (-> zettel
-      (merge changes)
-      (assoc :zettel/modified (java.util.Date.))))
+  (let [merged (-> zettel
+                   (merge changes)
+                   (assoc :zettel/modified (java.util.Date.)))]
+    (if (content-changed? changes)
+      (stamp-revision merged)
+      merged)))
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/knowledge/src/ai/miniforge/knowledge/zettel.clj]: Fixed. `update-zettel` now drops `:zettel/digest` and `:zettel/revision-id` from `changes` before the merge, then ALWAYS calls `stamp-revision` on the result. `stamp-revision` is idempotent on unchanged content (same content → same digest → same revision-id), so the operational-only-change semantics are preserved while spoofing is impossible. Two new tests pin the contract: `test-update-ignores-caller-supplied-derived-fields` (forged digest + forged revision-id are dropped, replaced with content-derived values; combined-with-real-content-change still rotates correctly from the projection, not the spoof). The plain-language consequence — legacy zettels without the derived fields backfill them on first update — is also tested in `test-update-backfills-derived-fields-on-legacy-zettel`. Resolving.

File: components/knowledge/src/ai/miniforge/knowledge/schema.clj [hunk 3]
```diff
@@ -100,7 +135,24 @@
...
    [:zettel/created inst?]
    [:zettel/modified {:optional true} inst?]
-   [:zettel/author [:string {:min 1}]]])         ; "user" or "agent:role"
+   [:zettel/author [:string {:min 1}]]           ; "user" or "agent:role"
+
+   ;; Revision-keyed identity (miniforge-fleet Decision 6). Optional in
+   ;; the schema so legacy zettels round-trip; new zettels stamp these
+   ;; via `zettel/create-zettel` and rotate via `zettel/update-zettel`.
+   [:zettel/revision-id {:optional true} uuid?]
+   [:zettel/digest      {:optional true} [:string {:min 64 :max 64}]]  ; SHA-256 hex
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/knowledge/src/ai/miniforge/knowledge/schema.clj]: Fixed. `:zettel/digest` is now `[:re #"^[0-9a-f]{64}$"]` — exactly the shape `zettel/compute-digest` produces (lowercase hex, 64 chars). Three new schema tests pin the boundary: `test-digest-schema-rejects-uppercase-hex`, `test-digest-schema-rejects-non-hex`, `test-digest-schema-rejects-wrong-length`. Stamped digests always validate (`test-digest-schema-accepts-canonical-hex`). Resolving.

File: components/knowledge/test/ai/miniforge/knowledge/zettel_revision_test.clj [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    (let [z1 (new-z :privacy/classification :internal :fleet/oss-version "1.0.0")
+          z2 (zettel/update-zettel z1 {:privacy/classification :restricted})
+          z3 (zettel/update-zettel z2 {:fleet/share-scope :team})
+          z4 (zettel/update-zettel z3 {:fleet/oss-version "1.0.1"})]
+      (is (= (:zettel/revision-id z1) (:zettel/revision-id z2)))
+      (is (= (:zettel/revision-id z2) (:zettel/revision-id z3)))
+      (is (= (:zettel/revision-id z3) (:zettel/revision-id z4)))
+      (is (= (:zettel/digest z1) (:zettel/digest z4))
+          "digest unchanged across operational-metadata updates"))))
+
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/knowledge/test/ai/miniforge/knowledge/zettel_revision_test.clj:103]: Both edge cases now have explicit coverage:

(1) **Spoof attempt via `update-zettel`** — `test-update-ignores-caller-supplied-derived-fields` covers two sub-cases. A pure spoof (forged digest + forged revision-id, no content change) is dropped + recomputed. A spoof combined with a real content change still rotates from the content projection, not the spoof.

(2) **Legacy zettel without the derived fields receiving an update** — `test-update-backfills-derived-fields-on-legacy-zettel` constructs a pre-Decision-6 zettel by hand and applies an operational-only change. The result has both `:zettel/digest` (64-char lowercase hex) and `:zettel/revision-id` (UUID) backfilled, plus the operational change landed. Behavior is intentional — `update-zettel` always re-stamps, so the system converges on a fully-stamped state without an explicit migration.

Resolving.
